### ISSUE TITLE
Lifecycle onPause/onStop/onDestory are never called

### ISSUE THREAD

**Issue by Syer10** [adrielcafe/voyager#42]
While testing my app with Voyager I noticed some of my AndroidView states were never being saved. Investigating that lead me to my LifecycleObserver's never filling the bundle contained in my rememberSavable variable. I then logged the states that the lifecycle observer goes through and noticed that only onCreate/OnStart/OnResume are ever called, and never the last 3. 

Here are the test composables I used
```kotlin
@Composable
private fun rememberLoggingLifecycleObserver(): LifecycleEventObserver {
    val bundle = rememberSaveable { Bundle() }
    return remember {
        LifecycleEventObserver { _, event ->
            when (event) {
                Lifecycle.Event.ON_CREATE -> {
                    logcat("Lifecycle") { "OnCreate" }
                    logcat("Lifecycle") { bundle.keySet().joinToString() }
                    bundle.clear()
                }
                Lifecycle.Event.ON_START -> logcat("Lifecycle") { "OnStart" }
                Lifecycle.Event.ON_RESUME -> logcat("Lifecycle") { "OnResume" }
                Lifecycle.Event.ON_PAUSE -> logcat("Lifecycle") { "OnPause" }
                Lifecycle.Event.ON_STOP -> {
                    logcat("Lifecycle") { "OnStop" }
                    bundle.putAll(bundleOf("Yes" to "1200"))
                }
                Lifecycle.Event.ON_DESTROY -> logcat("Lifecycle") { "OnDestroy" }
                else -> throw IllegalStateException()
            }
        }
    }
}
```

```kotlin
@Composable
fun SetupLoggerLifecycleObserver() {
    val loggerObserver = rememberLoggingLifecycleObserver()
    val lifecycle = LocalLifecycleOwner.current.lifecycle
    DisposableEffect(lifecycle) {
        lifecycle.addObserver(loggerObserver)
        onDispose {
            // lifecycle.removeObserver(loggerObserver)
        }
    }
}
```

**Comment by Syer10** [comment]
I was investigating this, and I think it may be because the lifecycle is not bound to the screen, but the activity. Even when I am using AndroidScreen screens, which I thought replaced the LocalLifecycleOwner composition local.

**Comment by Syer10** [comment]
I did some more investigating with it, and I tested providing the screen lifecycles to the content and noticed it was not working either. I found that the screens lifecycle never gets passed INITIALIZED. Here is the function I used to test it
```kotlin
@Composable
fun ScreenWithLifecycle(screen: Screen) {
    if (screen is AndroidScreen) {
        val lifecycleOwner by derivedStateOf { (screen.getLifecycleOwner() as AndroidScreenLifecycleOwner) }
        val lifecycleState by derivedStateOf { lifecycleOwner.lifecycle.currentState }
        LaunchedEffect(lifecycleState) {
            logcat("Lifecycle") { lifecycleState.name }
        }
        CompositionLocalProvider(
            LocalLifecycleOwner provides lifecycleOwner
        ) {
            screen.Content()
        }
    } else {
        screen.Content()
    }
}
```

**Comment by adrielcafe** [comment]
[1.0.0-rc02](https://github.com/adrielcafe/voyager/releases/tag/1.0.0-rc02) now provides a `LocalLifecycleOwner` and emits onPause/onStop/onDestroy states when using `AndroidScreen`.

I think it isn't enough yet. Seems I also need to provide a `LocalSaveableStateRegistry` but the Android implementation, `DisposableSaveableStateRegistry`, is internal. 

I'm still looking for a way to provide that, but in the meantime, can you confirm if the last modification solves your issue?

**Comment by Syer10** [comment]
Hmm, I don't think its working. The lifercycle observer is still isnt saving data into the bundle.

**Comment by miszmaniac** [comment]
I've encountered this issue. And due to that I can't handle ExoPlayer release() correctly in screen, when app get's paused. 
Any info about bugfix here?

**Comment by Syer10** [comment]
I have a PR to fix it #87.

The fix is also included in my fork of Voyager: https://github.com/Syer10/voyager
